### DISCUSSION TITLE
Issue with Sending OTP via SMS using MSG91 - Notification Data Pipeline Job Configuration

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Lern/Community#208] [Query]
Hello Team,

We are in the process of configuring SMS OTP functionality using the MSG91 provider. We have followed the configuration steps as outlined in the Sunbird developer guide: [SMS Configuration Guide](https://lern.sunbird.org/use/developer-guide/user-and-org-service/configuration/sms-configuration).

**Here’s what we have done so far:**
- Completed all configuration steps and set the following environment variables for the user-org service, notification service, and notification data pipeline job:
- sunbird_msg_91_auth=<authKey>
- sunbird_msg_sender=<SenderID registered with DLT>
- sms_gateway_provider=91SMS

**Issues we are facing:**

1. We are unable to find the sms_gateway_provider variable in the notification data pipeline job. Could you guide us on where this variable should be defined or set in the pipeline job configuration?
2. When testing the MSG91 API through Postman, we are receiving a success message; however, the OTP is not being delivered to the mobile number. We have validated that the sender ID and the auth key are correct.

**Request:**

- Please assist with setting the correct gateway configuration for the notification data pipeline job.
- Any troubleshooting steps or insights on why the OTP SMS is not being delivered despite the success message from the API would be greatly appreciated.

@komalm, @AmiableAnil , @amitpriyadarshi, @reshmi-nair , @pallakartheekreddy, @HarishGangula, @swayangjit @vinukumar-vs @rajeevsathish

**Comment by GayathriSrividya** [comment]
Hi @siddhishinde0723,
In case you are still facing the issue, here are the variables that need to be provided based on your service deployment. While the variable names might vary across different services, the values will remain the same.

For the **Userorg service**:
- `sunbird_msg_sender`
- `sunbird_msg_91_auth`

For the **Notification service**:
- `sunbird_notification_msg_default_sender`
- `sunbird_msg_91_auth`

For the **Notification Flink job** (in the Helm chart):
- `notification_msg_default_sender`
- `sunbird_msg_91_auth`

These variables should be mapped to the following in the `notification-job.conf` file:  
[notification-job.conf link](https://github.com/project-sunbird/sunbird-ed-installer/blob/release-6.0.0/helmcharts/learnbb/charts/flink/values.yaml#L804)
